web3/providers/ipc.py

```diff
@@ -204,10 +204,14 @@ def make_request(self, method: RPCEndpoint, params: Any) -> RPCResponse:
     def make_batch_request(
         self, requests: List[Tuple[RPCEndpoint, Any]]
     ) -> List[RPCResponse]:
-        self.logger.debug(f"Making batch request IPC. Path: {self.ipc_path}")
-        request_data = self.encode_batch_rpc_request(requests)
-        response = cast(List[RPCResponse], self._make_request(request_data))
-        return sort_batch_response_by_response_ids(response)
+        self._is_batching = True
```

Comment:
What do you think about creating a decorator that we can use on these methods that does the `try / finally` so we don't have to repeat this code all around? Just a thought, something like `@batching_context` / `@async_batching_context`?

Reply:
done!